UofT-Frosh-Orientation/orientation-website
# 694 complete status with points awarded

## **Contribution:**

- Added visual indication for completed missions along with the points awarded for the specific mission
- Removed instances for judging station on the missions page (since it won't be used anymore)
- Made enhancements to existing `transactions` property within `ScuntTeamModel` to keep track of existing missions
  - added some fixes to ScuntTeamServices, some async functions were missing awaits and some values were undefined and NaN
  - changed `calculatePoints`, since the `.map` function returned an array, resulting in a NaN value

### **Issue:**

Closes #694 

### **Accomplishments:**

- completed missions are indicated with points and a check
- backend works as expected

### **Visuals:** 

Any related visuals.

![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/86009011/aa061033-50a0-4344-a8a0-cb572ef76c62)
![image](https://github.com/UofT-Frosh-Orientation/orientation-website/assets/86009011/178b0478-f5af-480d-a15b-b5c47d6aaa1a)


## **Details:**

### **Expected Behavior:**

- On the frosh side once missions are completed, (they'll need to refresh their page), on the missions list page, they will see a green check beside completed missions and how many points were awarded

### **Testing Steps:**

Steps required to get expected behavior.

- Create a frosh account (attendingScunt = true) and leedur account with all scunt permissions
- Start scunt using the game settings, add missions and create teams
- Using the Scunt Judging Panel, assign points to the team that your frosh account is in
- On the missions list on the frosh side they will see the completed missions

### **Complications:**

### **Resources:**

Any additional resources used outside this in issue.
